go/arrow/csv/reader.go

```diff
@@ -721,6 +726,32 @@ func (r *Reader) parseDecimal256(field array.Builder, str string, prec, scale in
 	field.(*array.Decimal256Builder).Append(val)
 }
 
+func (r *Reader) parseList(field array.Builder, str string) {
+	if r.isNull(str) {
+		field.AppendNull()
+		return
+	}
+	if !(strings.HasPrefix(str, "{") || strings.HasSuffix(str, "}")) {
+		r.err = errors.New("invalid list format. should start with '{' and end with '}'")
+		field.AppendNull()
+		return
+	}
```

Comment:
The logic here isn't quite correct. It should be 

`if !strings.HasPrefix(str, "{") || !strings.HasSuffix(str, "}") {`

The current logic would only error if *both* are missing. Is there a unit test which tests the case where there is only a prefix "{" or only a suffix "}"?

Reply:
Good catch. Fixed and added tests.